amaranth/vendor/_gatemate.py

```diff
@@ -0,0 +1,90 @@
+from abc import abstractmethod
+from amaranth import *
+from amaranth.build import *
+from amaranth.lib.cdc import ResetSynchronizer
+
+
+__all__ = ["GateMatePlatform"]
+
+
+class GateMatePlatform(TemplatedPlatform):
+    """
+    Required tools:
+        * ``yosys``
+        * ``p_r``
+
+    The environment is populated by running the script specified in the environment variable
+    ``AMARANTH_ENV_GATEMATE``, if present.    
+    """
+
+    device = property(abstractmethod(lambda: None))
+    package = property(abstractmethod(lambda: None))
+
+    toolchain = "GateMate"
+
+    required_tools = [
+        "yosys",
+        "p_r",
+    ]
+
+    file_templates = {
+        **TemplatedPlatform.build_script_templates,
+        "{{name}}.v": r"""
+            /* {{autogenerated}} */
+            {{emit_verilog()}}
+        """,
+        "{{name}}.debug.v": r"""
+            /* {{autogenerated}} */
+            {{emit_debug_verilog()}}
+        """,
+        "{{name}}.ys": r"""
+            # {{autogenerated}}
+            {% for file in platform.iter_files(".v") -%}
+                read -sv {{get_override("read_verilog_opts")|options}} {{file}}
+            {% endfor %}
+            {{get_override("script_after_read")|default("# (script_after_read placeholder)")}}
+            synth_gatemate {{get_override("synth_opts")|options}} -top {{name}} -vlog {{name}}_synth.v
+            {{get_override("script_after_synth")|default("# (script_after_synth placeholder)")}}
+        """,
+        "{{name}}.ccf": r"""
+            # {{autogenerated}}
+            {% for port_name, pin_name, attrs in platform.iter_port_constraints_bits() -%}
+                Net "{{port_name}}" Loc = "{{pin_name}}"
+                {%- for constraint, value in attrs.items() -%}
+                | {{constraint}}={{value}}
+                {%- endfor -%};
+        {% endfor %}
+        """,
+        "{{name}}.sdc": r"""
+            # {{autogenerated}}
+            {% for net_signal, port_signal, frequency in platform.iter_clock_constraints() -%}
+                {% if port_signal is not none -%}
+                    create_clock -period {{1000000000/frequency}} [get_ports {{port_signal.name|tcl_quote}}]
+                {% endif %}
+            {% endfor %}
+        """,
+    }
+
+    command_templates = [
+        r"""
+        {{invoke_tool("yosys")}}
+            {{quiet("-q")}}
+            {{get_override("yosys_opts")|options}}
+            -l {{name}}.rpt
+            {{name}}.ys
+        """,
+        r"""
+        {{invoke_tool("p_r")}}
+            {{verbose("-v")}}
+            -i {{name}}_synth.v
+            -o {{name}}
+            -ccf {{name}}.ccf
+            -cCP > log/impl.log
```

Comment:
There is no log directory any more. The redirection should probably be something like this:
```suggestion
            -cCP
            > {{name}}.tim
```

The `.rpt`/`.tim` extension is something we've been using for Yosys based platforms and isn't set in stone, but it's nice to have consistency.

Reply:
Changed to 

```
        r"""
        {{invoke_tool("p_r")}}
            {{verbose("-v")}}
            -i {{name}}_synth.v
            -o {{name}}
            -ccf {{name}}.ccf
            -cCP > {{name}}.tim
        """,
```